### ISSUE TITLE
Whitespace issues

### ISSUE THREAD

**Issue by fbstj** [defunkt/pystache#6]
```
<ul>
{{#link}}
    <li><a href="{{url}}">{{text}}</a></li>
{{/link}}
</ul>
```

Its removing the 'newlineness' and the indentation (`\t`) of the `<li>`

Thanks for any help :) and thankyou very much for porting it in the first place,.

**Comment by defunkt** [comment]
Right now we don't preserve whitespace, but we really should. It's on The List.

**Comment by rafacv** [comment]
I've tried to approach the problem with this commit:
http://github.com/rafaelcv/pystache/commit/b767ea446d96d6e2e0f0c61ee3b2250aeed76278

I must say I'm not happy with produced code though and polishing is needed. It sucks even more because I didn't have the care of committing it to a clean branch instead of cluttering it on top of my commits to fix #8.

Anyway, commenting here just for the record.

**Comment by sebastien** [comment]
The quick fix I added is to change the section regexp to        

```
section = r"%(otag)s[\#|^]([^\}]*)%(ctag)s\s*(.+?\s*)%(otag)s/\1%(ctag)s"
```

**Comment by T1B0** [comment]
I've run in the kind of same issue as FallingBullets, 
with :

```
{{# css}}
    <link rel='stylesheet' type='text/css' href='{{{src}}}'/>
{{/ css}}
{{# script}}
<script type='text/javascript' src='{{{src}}}'></script>
{{/ script}}
```

i wanted :

```
    <link rel='stylesheet' type='text/css' href='/css/reset.css'/>
    <link rel='stylesheet' type='text/css' href='/css/mmm.css'/>
<script type='text/javascript' src='/js/jquery.js'></script>
<script type='text/javascript' src='/js/core.js'></script>
<script type='text/javascript' src='/js/mmm.js'></script>
```

 so i tweaked the regexp like this :

```
section = r"%(otag)s[\#|^]([^\}]*)%(ctag)s\n*([ \t\r\f\v]*.+\n*)\s*%(otag)s/\1%(ctag)s\s?"
self.section_re = re.compile(section % tags, re.M|re.S)
```

It does preserve indentation space and _some_ newlines, but is kind of flawed (especially with multiple newlines, or newlines before tag)

So that _work_ (barely) but tell me if you find anything fancyer

**Comment by gsong** [comment]
In `Template.render_sections()`'s `elif it:` block, instead of:

```
            replacer = ''.join(insides)
```

This instead:

```
            re_j = re.compile('(\s*)%s' % re.escape(inner), re.M)
            joiner = re_j.search(template).group(1)
            replacer = joiner.join(insides)
```

**Comment by reustle** [comment]
Hey @gsong, do you mind doing a pull request? I can as well. I'd love to see this fixed :)

**Comment by adamjernst** [comment]
Just ran into this too. :-(

**Comment by cjerdonek** [comment]
White space is important to me because I'm thinking about using pystache for Python source code templates.

However, it doesn't seem like there's a clear choice as to how white space should be handled if white space should be treated significantly.  For example, should white space before a template section item appear in the rendered version before every item, before just the first item, before every item but the first item, or before no items?  The similar question holds for white space after a template section item.

It seems like it might make sense to look at some real-world use cases first, and then see how easy or hard it is to work around the current behavior before deciding on any new behavior.

**Comment by cjerdonek** [comment]
I believe this issue has been fixed as part of pull request #27 (committed today to the development branch).
